app/frontend/src/features/communities/leaveReference/LeaveReferencePage.tsx

```diff
@@ -0,0 +1,122 @@
+import Hidden from "@material-ui/core/Hidden";
+import Alert from "components/Alert";
+import CircularProgress from "components/CircularProgress";
+import {
+  INVALID_REFERENCE_TYPE,
+  REFERENCE_TYPE_NOT_AVAILABLE,
+} from "features/communities/constants";
+import { useListAvailableReferences } from "features/communities/hooks";
+import ReferenceForm from "features/communities/leaveReference/ReferenceForm";
+import UserToReference from "features/communities/leaveReference/UserToReference";
+import { useUser } from "features/userQueries/useUsers";
+import { User } from "pb/api_pb";
+import { ReferenceType } from "pb/references_pb";
+import React from "react";
+import { useParams } from "react-router-dom";
+import makeStyles from "utils/makeStyles";
+
+const useStyles = makeStyles((theme) => ({
+  form: {
+    [theme.breakpoints.down("sm")]: {
+      margin: 0,
+      width: "100%",
+    },
+    flexGrow: 1,
+    margin: theme.spacing(2),
+    padding: theme.spacing(2),
+  },
+  root: {
+    [theme.breakpoints.up("md")]: {
+      display: "flex",
+      justifyContent: "space-around",
+    },
+  },
+}));
+
+export enum RefType {
```

Comment:
I think the API file is a better place to export this

Reply:
Also use an union type rather than enums. They're a pain in the arse since you have to do unnecessary type assertion in most cases to say it is `RefType` and also not type safe in some cases.

Union type gives you the same benefit but without all of those pain and it is type safe.